src/backend/optimizer/plan/createplan.c

```diff
@@ -640,6 +640,8 @@ use_physical_tlist(PlannerInfo *root, RelOptInfo *rel)
 static void
 disuse_physical_tlist(PlannerInfo *root, Plan *plan, Path *path)
 {
+	Assert(plan);
```

Comment:
Maybe it would be better to add a check before calling the disuse_physical_tlist function, as was done before its other call in the create_hashjoin_plan function?

Reply:
I think it's better to leave it here so that we won't have to check a plan in every calling function

Reply:
I decided to leave the check in disuse_physical_tlist because in your example there's no error in a case when outer_plan is null, it seems like it's expected and shouldn't be copied in other places